### PR TITLE
Upgrade http4k + Kotlin, adding tests, switch out templating engine for performance

### PR DESCRIPTION
**Pull request by daviddenton**

### CODE REVIEW

**Comment by daviddenton** [comment]
Sure - it turns out that the changes I made to replace UberJar with ShadowJar required some tweaking! :)

**Comment by daviddenton** [comment]
This is ready to merge now if someone would be so kind! 

Keep up the great work :)

**Comment by cjnething** [comment]
@daviddenton Thanks! I'll leave this to @nbrady-techempower or @msmith-techempower for review

File: frameworks/Kotlin/http4k/core/src/test/kotlin/Http4kBenchmarkServerTest.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.http4k.core.Method.GET
+import org.http4k.core.Request
+import org.http4k.format.Jackson.number
+import org.http4k.format.Jackson.obj
+import org.http4k.hamkrest.hasBody
+import org.junit.jupiter.api.Test
+import java.sql.Connection
+import java.sql.PreparedStatement
+
+class Http4kBenchmarkServerTest {
```
Comment (michaelhixson) [inline frameworks/Kotlin/http4k/core/src/test/kotlin/Http4kBenchmarkServerTest.kt]: We don't want to maintain any framework-specific functional tests in this repository.  Could you remove this file from the PR?

I made a similar request in a different PR recently and gave a longer explanation:  https://github.com/TechEmpower/FrameworkBenchmarks/pull/3895#discussion_r198972038

**Comment by daviddenton** [comment]
@michaelhixson done!

**Pull request merged**